server/data/data-jpa/src/test/java/org/apache/james/jpa/healthcheck/JPAHealthCheckTest.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.james.jpa.healthcheck;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.persistence.EntityManagerFactory;
+
+import org.apache.james.backends.jpa.JpaTestCluster;
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.core.healthcheck.ResultStatus;
+import org.apache.james.mailrepository.jpa.JPAUrl;
+import org.apache.openjpa.persistence.EntityManagerFactoryImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(JPAHealthCheckExtension.class)
+class JPAHealthCheckTest {
+
+    private JPAHealthCheck jpaHealthCheck;
+
+    @Test
+    void testWhenActive() {
+        final JpaTestCluster JPA_TEST_CLUSTER = JpaTestCluster.create(JPAUrl.class);
+        jpaHealthCheck = new JPAHealthCheck(JPA_TEST_CLUSTER.getEntityManagerFactory());
+        Result result = jpaHealthCheck.check();
+        ResultStatus healthy = ResultStatus.HEALTHY;
+        assertThat(result.getStatus()).as("Result %s status should be %s", result.getStatus(), healthy)
+                .isEqualTo(healthy);
+    }
+
+    @Test
+    void testWhenInactive() {
+        jpaHealthCheck = new JPAHealthCheck((EntityManagerFactory)
```

Comment:
How about adding a `JpaTestCluster::close method`?

Then we can be:
 - Creating the JPA test cluster
 - Creating the health check
 - Closing the JPA test cluster
 - Verify that the test is unhealthy

I think that would solve the NullPointerException encountered above.

Reply:
Will work on this later, thanks for regards

Reply:
I am still working in it. But now I managed to run the tests, a few tweaks in my IDE and environment. So pretty soon I am going to be able to fix that failure.